reconcile/utils/terraform_client.py

```diff
@@ -823,6 +823,19 @@ def validate_db_upgrade(
                     changed_fields=changed_fields,
                 )
 
+            db_response = self._aws_api.describe_rds_db_instance(
+                account_name, resource_name, region_name=region_name
+            )
+            replica_identifiers = db_response["DBInstances"][0].get(
+                "ReadReplicaDBInstanceIdentifiers"
+            )
+            db_identifier = db_response["DBInstances"][0].get("DBInstanceIdentifier")
+            if replica_identifiers is not None:
+                if len(replica_identifiers) > 0:
+                    raise RdsUpgradeValidationError(
+                        f"The primary database, {db_identifier}, has a read replica associated with it. Please remove the replica before upgrading the primary database"
+                    )
```

Comment:
this check now is for both major and minor version upgrade, doc only mentions major version upgrade, ask to remove read replica for a prod database is not good.

Since `terraform-resources` for RDS is being deprecated, we only need to handle it in erv2 module, and the better solution to upgrade instance with read replicas is to use new blue/green deployment.

Reply:
Thank you for your input! I'll go ahead and close this PR and will make the appropriate changes within the erv2 module